### PR TITLE
Replace the middleware URL

### LINKED ISSUE #80: Replace the middleware URL from `connect.woocommerce.com` to `api.woocommerce.com/integrations`
The `connect.woocommerce.com` server is due to be retired. The `api.woocommerce.com`  server now provides the connect service. The base URL will now be `api.woocommerce.com/integrations`. All other endpoints should now be accessible from the base URL.

## Endpoints

- [ ] `/login/square`
- [ ] `/auth/square`
- [ ] `/renew/square`
- [ ] `/revoke/square`
- [ ] `/login/squaresandbox`
- [ ] `/auth/squaresandbox`
- [ ] `/renew/squaresandbox`
- [ ] `/revoke/squaresandbox`

### PR DESCRIPTION
**Pull request by faisal-alvi**
### All Submissions:

* [ ] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:

This PR replaces the middleware URL from `connect.woocommerce.com` to `api.woocommerce.com/integrations`

Closes #80.

### Steps to test the changes in this Pull Request:

Test the connection and refresh the connection with production and sandbox environments.

### CODE REVIEW

**Comment by faisal-alvi** [comment]
We are blocked until the Connection URL works as expected and returns the correct redirect URL, which requires attention from the WooCommerce side. Read more -> https://github.com/woocommerce/woocommerce-square/issues/80#issuecomment-2006992829

**Comment by faisal-alvi** [comment]
Unblocked after https://github.com/woocommerce/woocommerce-square/issues/80#issuecomment-2014970404

**Review by dkotter — APPROVED** [review]
This looks good but we are using the `api-staging.woocommerce.com` URLs here (per https://github.com/woocommerce/woocommerce-square/issues/80#issuecomment-2014970404). This will need to be updated to the actual URL before we merge this in, after the connect service has been updated. Wondering if we hold off on testing this until that change has been made, so we don't have to test this one twice?

**Comment by faisal-alvi** [comment]
@dkotter agree.

@ibndawood the application on the staging works as expected, can you please change the redirect URI in production to `api.woocommerce.com` so we can perform the test and merge this PR?

**Comment by ibndawood** [comment]
Thank you for the confirmation. We'll set up rewrites in the `connect.woocommerce` server for the `square` endpoints. Once that is done, I'll update here, we can merge and make a release.

**Comment by ibndawood** [comment]
@faisal-alvi, @dkotter - We've setup rewrites in the `connect.woocommerce.com`. Now, all the requests are served from the `api.woocommerce.com`, making it backwards compatible. We are now good to release this PR.

**Comment by ibndawood** [comment]
Please make sure to use `api.woocommerce.com` in production and not `api-staging.woocommerce.com`.

File: includes/Handlers/Connection.php
Comment (dkotter) [inline includes/Handlers/Connection.php]:
```suggestion
	const CONNECT_URL_PRODUCTION = 'https://api.woocommerce.com/integrations/login/square';
```

File: includes/Handlers/Connection.php
Comment (dkotter) [inline includes/Handlers/Connection.php]:
```suggestion
	const CONNECT_URL_SANDBOX = 'https://api.woocommerce.com/integrations/login/squaresandbox';
```

File: includes/Handlers/Connection.php
Comment (dkotter) [inline includes/Handlers/Connection.php]:
```suggestion
	const REFRESH_URL_PRODUCTION = 'https://api.woocommerce.com/integrations/renew/square';
```

File: includes/Handlers/Connection.php
Comment (dkotter) [inline includes/Handlers/Connection.php]:
```suggestion
	const REFRESH_URL_SANDBOX = 'https://api.woocommerce.com/integrations/renew/squaresandbox';
```

**Comment by qasumitbagthariya** [comment]
### QA Update ✅ 

-------------

I have checked this issue in the `tweak/80` branch and the related issue is resolved and working as expected.

image

image


**Testing Environment** 

<Details>

- WordPress: 6.5
- Theme: Storefront 4.5.4
- Theme: Twenty Twenty-Four 1.1
- WooCommerce - 8.7.0
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- 

</Details>

Steps to Test- As mentioned in the PR description.
Test Results - It is working as expected.
Functional Demo / Screencast - 
Special Notes -  Ready for code review (Woo)
Testing Document status:
Cases related to this Issue/PR are added to the Critical Flow Wiki pages: 

- [ ] Yes
- [ ] Not Required/Applicable for this PR

**Review by qasumitbagthariya — APPROVED** [review]

**Review by wjrosa — APPROVED** [review]
Change looks good to me!

**Comment by faisal-alvi** [comment]
@qasumitbagthariya can you please confirm if the sync b/w WordPress and Square was also working as expected?

**Comment by qasumitbagthariya** [comment]
> @qasumitbagthariya can you please confirm if the sync b/w WordPress and Square was also working as expected?

@faisal-alvi cross check with the fix branch `tweak/80` Sync is working as expected 👍

**Review by qasumitbagthariya — APPROVED** [review]

**Pull request merged**
